### ISSUE TITLE
[BUG] sys/emunand ticket.db gets corrupted after manually reinstalling firmware

### ISSUE THREAD

**Issue by cooolgamer** [d0k3/GodMode9#857]
**So you want to report a bug?**
Hold on, there are ways you could make things easier for us:
* Give a clear description of the bug (what happened?).
* Give us clear steps to reproduce (when/how did it happen?).
* Give us info about your system (where did it happen?).
* A photograph or even a short video of the bug happening is always helpful!

**Info about your system**
Include this info to make our work easier:
* Console type (O3DS/N3DS)
* Anything special about your console? (defects, custom modifications,...)
* Bootloader (boot9strap/fastboot3ds)
* Did you chainload GodMode9 via Luma?
* Helpful hint: *if you followed the Guide, boot9strap is your bootloader and Luma is your chainloader.*

**Help yourself**
*Especially for any kind of boot issue ("GodMode9 doesn't boot")*, but also in many other cases these steps make a lot of sense and we will ask you to do them anyways:
* Check your SD card (using h2testw, f.e.) or try a different one (you wouldn't believe how common failing/fake SD cards are, and what kinds of bugs are caused by them).
* Switch to fastboot3DS using [https://github.com/d0k3/OpenFirmInstaller](OpenFirmInstaller).

**Have you actually read this?**
[X] I have read the information above

This is very specific and pretty rare, but when you want to manually do a ctrtransfer, so cleanup all ctrnand titles and install all firmware cias manually, there is a random chance that the ticket.db gets corrupted and everything fails to install because of this. I have no clue why this happens.
This was tested on emunand and sysnand on new and old 3ds.
This seems to happen after doing a lot of cia installation.

**Comment by Wolfvak** [comment]
@cooolgamer are you sure this happens "after doing a lot of cia installation"? Are you sure you're not installing something incorrectly? (just throwing out ideas)

It seems weird that it just breaks on its own after a certain amount of tickets are installed, especially if it's not consistent after doing things in the same order.

**Comment by cooolgamer** [comment]
Yes it's not consistent because if I repair the ticket.db, repeat the steps over, it works.

Also note that this happens during the cias installation and the next ones always fails.

Other note, titles are system apps (NAND), I don't know if the same thing can happen for SD titles.

It doesn't only happen to me, someone else also got this issue, I also tested on different systems, same result, there's a random chance of ticket.db being corrupted while installing a lot of NAND cia files.

**Comment by d0k3** [comment]
The issue is most likely in `bdri.c`, which I'm not really familiar with. If someone is willing to test, here's a potential fix. Nothing bad should happen, but be careful, and have backups of everything. 
[GodMode9-v2.2.3-2-g57932315-20260509121136.zip](https://github.com/user-attachments/files/27550138/GodMode9-v2.2.3-2-g57932315-20260509121136.zip)

**Comment by d0k3** [comment]
Thanks @redthunderpower1 for the quick and thorough test. Further testing is highly appreciated, please let me know if problems remain.
